packages/matter.js/src/common/ValidationError.ts

```diff
@@ -7,16 +7,42 @@
 import { StatusCode, StatusResponseError } from "../protocol/interaction/StatusCode.js";
 
 /**
- * Error thrown when Data validation fails. Please check the provided data.
- * We encode this as StatusResponseError because in most places we want to return a StatusResponseError anyway.
+ * Base class for validation errors and used when Data validation fails. Please check the provided data.
+ * We encode this as StatusResponseError that returns a ConstrainError which is suitable for most
+ * "invalid/out-of-bounds values" cases. If you can use any of the more detailed errors derived from this class please
+ * do so!
  */
 export class ValidationError extends StatusResponseError {
     public constructor(
         message: string,
         public fieldName?: string,
+        errorCode = StatusCode.ConstraintError,
     ) {
-        super(message, StatusCode.ConstraintError);
+        super(message, errorCode);
 
-        this.message = `(Validation/${StatusCode.ConstraintError}) ${message}`;
+        this.message = `(Validation/${this.code}) ${message}`;
```

Comment:
Do you think we should include class name and (if applicable) status code in the error formatter?  I was thinking about that in some of my stuff.  We've got a couple different conventions for these so would be nice to normalize

Reply:
I adjusted